js/phemex.js

```diff
@@ -2932,12 +2934,63 @@ module.exports = class phemex extends Exchange {
         };
     }
 
+    async modifyMarginHelper (symbol, amount, addOrReduce, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol': market['id'],
+            'posBalanceEv': this.toEv (amount, market),
+        };
+        const response = await this.privatePostPositionsAssign (this.extend (request, params));
+        //
+        //     {
+        //         "code": 0,
+        //         "msg": "",
+        //         "data": "OK"
+        //     }
+        //
+        return this.extend (this.parseModifyMargin (response, market), {
+            'amount': amount,
+            'type': addOrReduce,
+        });
+    }
+
+    parseMarginStatus (status) {
+        const statuses = {
+            '0': 'ok',
+        };
+        return this.safeString (statuses, status, status);
+    }
+
+    parseModifyMargin (data, market = undefined) {
+        const codeCurrency = market['inverse'] ? 'base' : 'quote';
+        return {
+            'info': data,
+            'type': undefined,
+            'amount': undefined,
+            'code': market[codeCurrency],
```

Comment:
We must use safeMarket here, according to the signature of this method the `market` may be `undefined`, despite that we always supply it.